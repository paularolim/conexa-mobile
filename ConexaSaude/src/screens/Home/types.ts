import { Appointment } from '@hooks/useListAppointments/types';

import { appointments } from './mock';

export interface HomeViewProps {
  appointments: Appointment[];
  name: string;
  refreshing: boolean;
  handlePressAppointment: () => void;
  onRefresh: () => void;
}

export type MockAppointments = (typeof appointments.data)[0];
