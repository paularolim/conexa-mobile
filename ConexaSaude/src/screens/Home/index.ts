export * from './Home';
export * from './HomeView';
export * from './types';
export * from './useHomeViewModel';
