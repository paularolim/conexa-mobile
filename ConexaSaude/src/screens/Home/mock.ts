export const appointments = {
  data: [
    {
      id: 16,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Diego Senna',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
    {
      id: 11,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Rafael Braga',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
    {
      id: 12,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Igor Silva',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
    {
      id: 13,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Igor Costa',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
    {
      id: 14,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Deiveson Barreto',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
    {
      id: 15,
      medico: {
        id: 1,
        nome: 'Dr. Daniel Vieira',
        email: null,
      },
      paciente: 'Adriano Sena',
      dataConsulta: '2020-04-23 08:30',
      observacao: 'Exemplo de consulta',
    },
  ],
};
