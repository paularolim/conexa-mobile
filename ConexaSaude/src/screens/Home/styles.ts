import { View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(View)`
  flex: 1;
  padding: 24px 24px 0px 24px;
  gap: 20px;
  background-color: ${({ theme }) => theme.colors.white};
`;

export const Header = styled(View)`
  padding-bottom: 16px;
  gap: 8px;
  background-color: ${({ theme }) => theme.colors.white};
`;

export const Title = styled(Text)`
  font-size: 20px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.gray};
`;

export const Separator = styled(View)`
  height: 10px;
`;

export const Footer = styled(View)`
  height: 30px;
`;
