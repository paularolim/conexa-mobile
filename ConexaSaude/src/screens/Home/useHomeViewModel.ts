import { useNavigation } from '@react-navigation/native';
import { useCallback, useState } from 'react';

import { useAuth } from '@hooks/useAuth';

export function useHomeViewModel() {
  const [refreshing, setRefreshing] = useState(false);

  const { navigate } = useNavigation();
  const { name } = useAuth();

  const onRefresh = useCallback(() => {
    setRefreshing(true);
    setTimeout(() => {
      setRefreshing(false);
    }, 2000);
  }, []);

  const handlePressAppointment = useCallback(() => {
    navigate('Appointment' as never);
  }, [navigate]);

  return { handlePressAppointment, refreshing, onRefresh, name };
}
