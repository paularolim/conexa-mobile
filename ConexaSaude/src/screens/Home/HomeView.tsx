import { useCallback } from 'react';
import { FlatList, ListRenderItemInfo, RefreshControl } from 'react-native';

import { AppointmentCard } from '@components/AppointmentCard';
import { Appointment } from '@hooks/useListAppointments/types';

import { Container, Footer, Header, Separator, Title } from './styles';
import { HomeViewProps } from './types';

export function HomeView({
  name,
  refreshing,
  onRefresh,
  appointments,
  handlePressAppointment,
}: HomeViewProps) {
  const headerItem = useCallback(
    () => (
      <Header>
        <Title>Ol√° {name}!</Title>
        <Title>Agenda</Title>
      </Header>
    ),
    [name],
  );

  const footerItem = useCallback(() => <Footer />, []);

  const separatorItem = useCallback(() => <Separator />, []);

  const renderItem = useCallback(
    ({ item }: ListRenderItemInfo<Appointment>) => (
      <AppointmentCard date={item.date} patient={item.patient} onPress={handlePressAppointment} />
    ),
    [handlePressAppointment],
  );

  return (
    <Container>
      <FlatList
        data={appointments || []}
        keyExtractor={(item) => `${item.id}`}
        renderItem={renderItem}
        ItemSeparatorComponent={separatorItem}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={[0]}
        ListHeaderComponent={headerItem}
        ListFooterComponent={footerItem}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} />}
      />
    </Container>
  );
}
