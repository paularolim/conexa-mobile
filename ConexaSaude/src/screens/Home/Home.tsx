import { HomeView } from './HomeView';
import { ScreenProps } from './types';
import { useHomeViewModel } from './useHomeViewModel';

export function Home(screenProps: ScreenProps) {
  const props = useHomeViewModel(screenProps);
  return <HomeView {...props} />;
}
