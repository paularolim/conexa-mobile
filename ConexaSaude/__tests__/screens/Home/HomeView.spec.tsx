import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { HomeView } from '@screens/Home/HomeView';

import { Wrapper } from '../../Wrapper';

describe('HomeView', () => {
  it('should render screen correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <HomeView
            appointments={[
              { date: 'mocked_date', id: 0, observation: 'mocked_info', patient: 'mocked_patient' },
              { date: 'mocked_date', id: 1, observation: 'mocked_info', patient: 'mocked_patient' },
            ]}
            name="Dr. John Doe"
            refreshing={false}
            loading={false}
            error={false}
            handlePressAppointment={jest.fn()}
            handleCreateAppointment={jest.fn()}
            onRefresh={jest.fn()}
          />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render screen correctly on loading', () => {
    const { getByTestId } = render(
      <Wrapper>
        <HomeView
          appointments={[]}
          name="Dr. John Doe"
          refreshing={false}
          loading
          error={false}
          handlePressAppointment={jest.fn()}
          handleCreateAppointment={jest.fn()}
          onRefresh={jest.fn()}
        />
      </Wrapper>,
    );
    expect(getByTestId('activity-indicator')).toBeTruthy();
  });

  it('should render screen correctly on error', () => {
    const { getByTestId } = render(
      <Wrapper>
        <HomeView
          appointments={[]}
          name="Dr. John Doe"
          refreshing={false}
          loading={false}
          error
          handlePressAppointment={jest.fn()}
          handleCreateAppointment={jest.fn()}
          onRefresh={jest.fn()}
        />
      </Wrapper>,
    );
    expect(getByTestId('error-message')).toBeTruthy();
  });

  it('should render screen correctly on error', () => {
    const { getByTestId } = render(
      <Wrapper>
        <HomeView
          appointments={[]}
          name="Dr. John Doe"
          refreshing={false}
          loading={false}
          error={false}
          handlePressAppointment={jest.fn()}
          handleCreateAppointment={jest.fn()}
          onRefresh={jest.fn()}
        />
      </Wrapper>,
    );
    expect(getByTestId('empty-message')).toBeTruthy();
  });
});
