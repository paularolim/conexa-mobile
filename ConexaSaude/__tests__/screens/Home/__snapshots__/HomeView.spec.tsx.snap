// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView should render screen correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "gap": 20,
        "paddingBottom": 0,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      [
        {
          "date": "mocked_date",
          "id": 0,
          "observation": "mocked_info",
          "patient": "mocked_patient",
        },
        {
          "date": "mocked_date",
          "id": 1,
          "observation": "mocked_info",
          "patient": "mocked_patient",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={
      [
        0,
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        collapsable={false}
        onLayout={[Function]}
      >
        <View
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "gap": 8,
                "paddingBottom": 16,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                },
                {
                  "color": "#707683",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Ol√° 
            Dr. John Doe
            !
          </Text>
          <Text
            style={
              [
                {
                  "color": "#000000",
                },
                {
                  "color": "#707683",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Agenda
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2CFE0",
              "borderRadius": 12,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "gap": 12,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgCalendar"
              height={28}
              stroke="#C2CFE0"
              width={28}
            />
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "color": "#707683",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Data
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                  ]
                }
              >
                mocked_date
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "gap": 10,
                },
              ]
            }
          >
            <View
              primary={true}
              style={
                [
                  {
                    "backgroundColor": "#F28080",
                    "borderRadius": 2,
                    "height": 5,
                    "width": 80,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#C2CFE0",
                    "borderRadius": 2,
                    "height": 5,
                    "width": 80,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#C2CFE0",
                    "borderRadius": 20,
                    "height": 40,
                    "overflow": "hidden",
                    "width": 40,
                  },
                ]
              }
            >
              <svg
                data-file-name="SvgUser"
                height={40}
                stroke="#F28080"
                width={40}
              />
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              mocked_patient
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 10,
              },
            ]
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2CFE0",
              "borderRadius": 12,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "gap": 12,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                },
              ]
            }
          >
            <svg
              data-file-name="SvgCalendar"
              height={28}
              stroke="#C2CFE0"
              width={28}
            />
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "color": "#707683",
                      "fontSize": 12,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Data
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                  ]
                }
              >
                mocked_date
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "gap": 10,
                },
              ]
            }
          >
            <View
              primary={true}
              style={
                [
                  {
                    "backgroundColor": "#F28080",
                    "borderRadius": 2,
                    "height": 5,
                    "width": 80,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#C2CFE0",
                    "borderRadius": 2,
                    "height": 5,
                    "width": 80,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "gap": 8,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#C2CFE0",
                    "borderRadius": 20,
                    "height": 40,
                    "overflow": "hidden",
                    "width": 40,
                  },
                ]
              }
            >
              <svg
                data-file-name="SvgUser"
                height={40}
                stroke="#F28080"
                width={40}
              />
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                ]
              }
            >
              mocked_patient
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            [
              {
                "height": 30,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
